src/Build/BuildCheck/Infrastructure/BuildCheckManagerProvider.cs

```diff
@@ -340,6 +355,19 @@ public void RemoveThrottledChecks(ICheckContext checkContext)
             }
         }
 
+        public void RemoveChecks(List<CheckWrapper> checkToRemove, ICheckContext checkContext)
+        {
+            foreach (CheckWrapper check in checkToRemove)
+            {
+                var checkFactory = _checkRegistry.Find(c => c.MaterializedCheck == check);
+                if (checkFactory is not null)
+                {
+                    checkContext.DispatchAsCommentFromText(MessageImportance.High, $"Dismounting check '{check.Check.FriendlyName}'");
```

Comment:
Let's put the reason - something like 'Dismounting check 'xyz' due to unhandled exception thrown by the check.'